src/main/java/com/google/api/generator/gapic/composer/rest/HttpJsonServiceStubClassComposer.java

```diff
@@ -240,12 +240,12 @@ protected Expr createTransportSettingsInitExpr(
   protected List<AnnotationNode> createClassAnnotations(Service service) {
     List<AnnotationNode> annotations = super.createClassAnnotations(service);
 
-    annotations.add(
-        AnnotationNode.builder()
-            .setType(FIXED_TYPESTORE.get("BetaApi"))
-            .setDescription(
-                "A restructuring of stub classes is planned, so this may break in the future")
-            .build());
+    TypeNode betaApiType = FIXED_TYPESTORE.get("BetaApi");
+
+    if (annotations.stream().noneMatch(a -> betaApiType.equals(a.type()))) {
```

Comment:
I think this part makes sense, we are adding the `BetaApi` annotation to classes if they have `beta` or `alpha` in their package name currently, but we want to add it to any HttpJson related class even if the proto says it's not Beta anymore. Can we add a test case for it though? Like removing 'beta' from the package name [here](https://github.com/vam-google/gapic-generator-java/blob/c8672801bd489740dae413cf4f913ae466330775/src/test/proto/echo_grpcrest.proto#L29).

Reply:
Having beta in echo_grpcrest.proto tests this specific block of not applying BetaApi twice. Having the whole new set of golden files and input protos to test the small case of still adding "beta" annotation for beta apis in httpjson case (we already have it tested for grpc case as an integration test for `redis/v1beta1` api) seems like not worth it (too many test files to test a very minor thing). Plus, once we have REGAPCI pushed, it will add rest transport to existing integration tests and this case will be tested automatically by integration tests.

Reply:
Add a unit test for this method should be pretty straightforward if you think adding a file is not worth it, since both the input and output of this method are just model class created in this repo.

Reply:
Composer classes are not tested by testing individual methods in them, instead they are tested by fully generating the file they are responsible for (in this case it would be `HttpJson<Service>Stub.java`). So to test generation of a stub class under a different package we need: new input proto file (with different package), new .golden file to match output, new method in `HttpJsonServiceStubClassComposerTest` class to trigger that check. It is a trivial thing to do, bu honestly I don't think it is  a right thing to do, because it would bloat our testing infrastructure - we can't make the whole new set of input protos and goldens to test every tiny feature like this one.

Reply:
I think you have a very valid point and I completely agree: It's not a right thing to create a new proto and golden file for every tiny feature. So we should change the way we are testing composers, for complicated things that are hard to write unit tests, we can continue to test them using protos/goldens, but for tiny things like this we should probably start doing true unit testing - test the method directly. I have seen a lot of code missing unit tests in this repo and I think we can make it better.